lib/transport/selenium-webdriver/method-mappings.js

```diff
@@ -124,76 +124,133 @@ module.exports = class MethodMappings {
-        switchToWindow(handle) {
-          return TransportActions.post({
-            path: '/window',
-            data: {
-              name: handle
-            }
-          });
+        async switchToWindow(handle) {
+          await this.driver.switchTo().window(handle);
+
+          return {
+            value: null
+          };
         },
 
-        closeWindow() {
-          return TransportActions.delete('/window');
+        async closeWindow() {
+          await this.driver.close();
+
+          return {
+            value: null
+          };
         },
 
-        getWindowHandle: '/window_handle',
+        async getWindowHandle() {
+          const value = await this.driver.getWindowHandle();
+
+          return {
+            value
+          };
+        },
 
-        getAllWindowHandles: '/window_handles',
+        async getAllWindowHandles() {
+          const value = await this.driver.getAllWindowHandles();
 
-        getWindowPosition(windowHandle) {
-          return `/window/${windowHandle}/position`;
+          return {
+            value
+          };
         },
 
-        maximizeWindow(windowHandle) {
-          return TransportActions.post(`/window/${windowHandle}/maximize`);
+        async getWindowPosition() {
+          const { x, y } = await this.driver.manage().window().getRect();
+
+          return {
+            value: {
+              x,
+              y
+            }
+          };
         },
 
-        minimizeWindow() {
-          return TransportActions.post('/window/minimize');
+        async maximizeWindow(windowHandle) {
+          await this.driver.manage().window().maximize();
+
+          return {
+            value: null
+          };
         },
 
-        fullscreenWindow() {
-          return TransportActions.post('/window/fullscreen');
+        async minimizeWindow() {
+          await this.driver.manage().window().minimize();
+
+          return {
+            value: null
+          };
         },
 
-        openNewWindow() {
-          return TransportActions.post('/window/new');
+        async fullscreenWindow() {
+          await this.driver.manage().window().fullscreen();
+
+          return {
+            value: null
+          };
         },
 
-        setWindowPosition(windowHandle, offsetX, offsetY) {
-          return TransportActions.post({
-            path: `/window/${windowHandle}/position`,
-            data: {
-              x: offsetX,
-              y: offsetY
-            }
-          });
+        async openNewWindow() {
+          // return TransportActions.post('/window/new');
+          await this.driver.switchTo().newWindow('window');
+
+          return {
+            value: null
+          };
         },
 
-        getWindowSize(windowHandle) {
-          return `/window/${windowHandle}/size`;
+        //Kept windowHandle as params for backward comptability.
+        async setWindowPosition(windowHandle, offsetX, offsetY) {
```

Comment:
you can simply use `x` and `y` as params.

Reply:
Will do the change 👍